fix(shared): mcctl playit start fails when container has never been started

## Summary

`mcctl playit start` fails with `[ERROR] playit.gg is not enabled` even when `PLAYIT_SECRET_KEY` is properly configured in `.env` and `playitEnabled: true` is set in `.mcctl.json`.

## Root Cause

In `platform/services/shared/src/docker/index.ts`, `getPlayitAgentStatus()` calculates `enabled` as:

```typescript
enabled: containerExists(containerName) && secretKeyConfigured,
```

This creates a chicken-and-egg problem: the container doesn't exist because it has never been started, but it can't be started because `enabled` is `false` due to `containerExists()` returning `false`.

## Solution

Change the `enabled` logic to check `.mcctl.json` `playitEnabled` flag OR `secretKeyConfigured`, independent of container existence:

```typescript
enabled: playitEnabled || secretKeyConfigured,
```

Container existence should not be part of the configuration state check - it's a runtime state.

## Files to Modify

- `platform/services/shared/src/docker/index.ts` - Fix `enabled` logic in `getPlayitAgentStatus()`
- `platform/services/shared/tests/` - Add/update tests for the fix

## Branch Strategy

- Type: bugfix
- Branch: `bugfix/337-playit-start-enabled-check`

## Acceptance Criteria

- [ ] `mcctl playit start` works when SECRET_KEY is configured but container never started
- [ ] `enabled` field reflects configuration state, not container existence
- [ ] Tests pass

--- a/platform/services/cli/tests/unit/commands/playit.test.ts
+++ b/platform/services/cli/tests/unit/commands/playit.test.ts
@@ -136,6 +136,29 @@ describe('mcctl playit subcommand', () => {
       expect(exitCode).toBe(1);
       expect(shared.startPlayitAgent).not.toHaveBeenCalled();
     });
+
+    it('should start when key is configured but container never created (first start)', async () => {
+      // This is the chicken-and-egg scenario: container doesn't exist yet
+      // because it has never been started, but start should still work
+      createInitializedPlatform(true, 'test-key-123');
+
+      vi.mocked(shared.getPlayitAgentStatus).mockResolvedValueOnce({
+        enabled: true, // Should be true based on config, not container existence
+        agentRunning: false,
+        secretKeyConfigured: true,
+        containerStatus: 'not_found', // Container never created
+      });
+
+      vi.mocked(shared.startPlayitAgent).mockResolvedValueOnce(true);
+
+      const exitCode = await playitCommand({
+        root: testRoot,
+        subCommand: 'start',
+      });
+
+      expect(exitCode).toBe(0);
+      expect(shared.startPlayitAgent).toHaveBeenCalled();
+    });
   });
 
   describe('playit stop', () => {

--- a/platform/services/shared/src/docker/index.ts
+++ b/platform/services/shared/src/docker/index.ts
@@ -981,6 +981,18 @@ export async function getPlayitAgentStatus(): Promise<import('../types/index.js'
     }
   }
 
+  // Check if playit is enabled in .mcctl.json config
+  const configPath = join(platformRoot, '.mcctl.json');
+  let playitEnabled = false;
+  if (existsSync(configPath)) {
+    try {
+      const config = JSON.parse(readFileSync(configPath, 'utf-8'));
+      playitEnabled = config.playitEnabled === true;
+    } catch {
+      // Ignore parse errors
+    }
+  }
+
   // Get uptime if running
   let uptime: string | undefined;
   let uptimeSeconds: number | undefined;
@@ -994,7 +1006,7 @@ export async function getPlayitAgentStatus(): Promise<import('../types/index.js'
   }
 
   return {
-    enabled: containerExists(containerName) && secretKeyConfigured,
+    enabled: playitEnabled || secretKeyConfigured,
     agentRunning,
     secretKeyConfigured,
     containerStatus: status,

